eclipse-collections/src/main/java/org/eclipse/collections/impl/multimap/AbstractSynchronizedPutMultimap.java

```diff
@@ -92,6 +92,21 @@ public boolean put(K key, V value)
         }
     }
 
+    @Override
+    public Iterable<? extends V> getIfAbsentPut(K key, Iterable<? extends V> value)
+    {
+        boolean hasKey = this.containsKey(key);
+        if (!hasKey)
+        {
+            synchronized (value)
```

Comment:
I think they followed the implementation of `put()`
If I am not mistaken the backing Map for SynchronizedPutMultimap is a ConcurrentHashmap.So, the override should be to delegate to backing map.
```
    @Override
    protected MutableMap<K, C> createMap()
    {
        return ConcurrentHashMap.newMap();
    }
```
```
   @Override
    public Iterable<? extends V> getIfAbsentPut(K key, Iterable<? extends V> value)
    {
        return this.map.getIfAbsentPut(key, value);
    }
```

Reply:
No because value is an Iterable, so casting it to a Collection is wrong. I think this implementation will work in AbstractMutableMultimap and will be as efficient as possible:
```
    @Override
    public Iterable<? extends V> getIfAbsentPut(K key, Iterable<? extends V> value)
    {
        C collection = this.getIfAbsentPutCollection(k);
        if(collection.isEmpty())
        {
             Iterate.addAllTo(values, collection)
        }
        return collection;
    }
```

Reply:
I have implemented this in AbstractMutableMultimap, but the implementation in AbstractSyncrhonizedPutMultiMap is producing an error.

Reply:
You needn't override it in synchronized because the backing map is a concurrentmap so the synchronization will be handled by it

Reply:
Okay. I removed the implementation within AbstractMutableMultmap, but I am still getting a Cannot resolve method in the AbstractSyncrhonizedPutMultimap with 
return this.map.getIfAbsentPut(key, value);

Reply:
Why would you remove it from AbstractMutableMultimap? Keep it there, remove it from synchronized

Reply:
Oh, my bad. I got seriously confused. Correcting this and then going to commit again.

Reply:
@nikhilnanivadekar without an override, it's not threadsafe at all. Even operations that do a lot less like put() are overridden.

Reply:
What should be my next steps going forward with this? Should I alter the implementation of pulAll() as well?